olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp

```diff
@@ -236,156 +361,209 @@ TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyFound) {
-                        olp::http::Network::HeaderCallback,
-                        olp::http::Network::DataCallback) {
+      .WillByDefault(testing::WithoutArgs([]() {
         ADD_FAILURE()
             << "Should not be called. Previous request was cancelled.";
         constexpr auto unused_request_id = 5;
-        return olp::http::SendOutcome(unused_request_id);
-      });
+        return http::SendOutcome(unused_request_id);
+      }));
 
   ApiLookupClient lookup_client(kHrn, settings_);
   repository::CatalogRepository repository(kHrn, settings_, lookup_client);
-  auto response = repository.GetLatestVersion(request, context);
+  const auto response = repository.GetLatestVersion(request, context);
 
-  ASSERT_FALSE(response.IsSuccessful());
-  EXPECT_EQ(olp::client::ErrorCode::Cancelled,
-            response.GetError().GetErrorCode());
+  ASSERT_FALSE(response);
+  EXPECT_EQ(client::ErrorCode::Cancelled, response.GetError().GetErrorCode());
 }
 
 TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyUserCancelled2) {
-  olp::client::CancellationContext context;
+  client::CancellationContext context;
 
-  auto request = read::CatalogVersionRequest();
+  const auto request = read::CatalogVersionRequest();
 
   ON_CALL(*network_, Send(IsGetRequest(kLookupMetadata), _, _, _, _))
-      .WillByDefault(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(
-                                            olp::http::HttpStatusCode::OK),
-                                        kResponseLookupMetadata));
+      .WillByDefault(ReturnHttpResponse(
+          http::NetworkResponse().WithStatus(http::HttpStatusCode::OK),
+          kResponseLookupMetadata));
 
   ON_CALL(*network_, Send(IsGetRequest(kLatestCatalogVersion), _, _, _, _))
-      .WillByDefault([&](olp::http::NetworkRequest, olp::http::Network::Payload,
-                         olp::http::Network::Callback,
-                         olp::http::Network::HeaderCallback,
-                         olp::http::Network::DataCallback) {
+      .WillByDefault(testing::WithoutArgs([&]() {
         std::thread([&]() { context.CancelOperation(); }).detach();
 
         constexpr auto unused_request_id = 10;
-        return olp::http::SendOutcome(unused_request_id);
-      });
+        return http::SendOutcome(unused_request_id);
+      }));
 
   ApiLookupClient lookup_client(kHrn, settings_);
   repository::CatalogRepository repository(kHrn, settings_, lookup_client);
-  auto response = repository.GetLatestVersion(request, context);
+  const auto response = repository.GetLatestVersion(request, context);
 
-  ASSERT_FALSE(response.IsSuccessful());
-  EXPECT_EQ(olp::client::ErrorCode::Cancelled,
-            response.GetError().GetErrorCode());
+  ASSERT_FALSE(response);
+  EXPECT_EQ(client::ErrorCode::Cancelled, response.GetError().GetErrorCode());
+}
+
+TEST_F(CatalogRepositoryTest, AsyncGetLatestVersionOnlineOnlyUserCancelled2) {
+  const auto request = read::CatalogVersionRequest();
```

Comment:
A default fetch option is OnlineIfNotFound. But the test name contains `OnlineOnly`